pkg/pillar/scripts/device-steps.sh

```diff
@@ -428,23 +477,13 @@ echo "$(date -Ins -u) Done starting EVE version: $(cat /run/eve-release)"
 # and dump any diag information
 while true; do
     access_usb
-    # Check if NTP server changed
+    # Check if NTP servers changed
     # Note that this really belongs in a separate ntpd container
-    ns=$(get_ntp_server)
-    if [ -n "$ns" ] && [ "$ns" != "$NTPSERVER" ] && [ -f /run/ntpd.pid ]; then
-        echo "$(date -Ins -u) NTP server changed from $NTPSERVER to $ns"
-        NTPSERVER="$ns"
-        ntpd_pid="$(cat /run/ntpd.pid)"
-        kill "$ntpd_pid"
-        # Wait for it to go away before restarting
-        while kill -0 "$ntpd_pid"; do
-            echo "$(date -Ins -u) NTP server $ntpd_pid still running"
-            sleep 3
-        done
-        echo "$(date -Ins -u) ntpd -g -p $NTPSERVER"
-        /usr/sbin/ntpd -g -p "$NTPSERVER"
-        ret_code=$?
-        echo "$(date -Ins -u) ntpd: $ret_code"
+    ns=$(get_ntp_servers)
+    if [ -n "$ns" ] && [ "$ns" != "$NTPSERVERS" ]; then
+        echo "$(date -Ins -u) NTP server changed from \"$NTPSERVERS\" to \"$ns\", restarting NTP daemon"
+        # Restart NTP
+        restart_ntp_daemon
     fi
     sleep 300
```

Comment:
In docs we should probably mention that it may take up to 300 secs for NTP config change to take effect.